ksql-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java

```diff
@@ -66,14 +66,15 @@ public KsqlEngine(
       final ServiceContext serviceContext,
       final ProcessingLogContext processingLogContext,
       final FunctionRegistry functionRegistry,
-      final String serviceId
+      final String serviceId,
```

Comment:
nit: I think we should couple `serviceId` and `custommetricsTags` into a `ServiceInfo` class so that piping this type of information down will be easier in the future (also I imagine that `serviceId` might be valuable for metrics as well)